[charts] Make sqrt size mapping the default (#22683)
mui/mui-x: docs/data/charts/scatter/scatter.md

The `sizeMap` supports the same `continuous`, `piecewise`, and `ordinal` types as `colorMap`, but it maps values to a marker radius in pixels.
A series can set both `colorAxisId` and `sizeAxisId` to style points by two values at once.

> :::info

Reviewer: Should this be a section+demo instead of a "info" block?

> By default the `sizeMap` with type `'continuouse'` maps values to radius using a square root scale.

Reviewer: ```suggestion
By default the `sizeMap` property with type `'continuous'` transforms values into radius using a square root scale.
```
